xapian-letor/ranker/listmle_ranker.cc

```diff
@@ -71,7 +72,7 @@ static vector<double>
 calculate_gradient(const vector<FeatureVector>& sorted_feature_vectors,
 		   const vector<double>& new_parameters)
 {
-    //need to be optimized,how to get the length of the features
+    // Need to be optimized,how to get the length of the features
```

Comment:
This sort of "todo note" comment is better marked with a "FIXME" tag.  These are easy to search for (e.g. `git grep FIXME`) and stand out more when reading the code (some editors even colour the "FIXME" text specially, but just as a all-caps word it tends to stand out).

`xapian-check-patch` will warn about adding a "FIXME", but that's a warning not an error and the CI won't fail because of it.  The reason for it is help ensure that "FIXME" notes added during development with the intention of addressing them before pushing the patch haven't just been overlooked (long-term FIXMEs are OK, things you meant to fix as part of development aren't).

I'm not clear what this note is actually about though - the implementation of `get_fcount()` just seems to be a call to `std::vector::size()`, which is a cheap operation.  How could this be optimised?